shell/packages/sandstorm-db/db.js

```diff
@@ -560,6 +566,52 @@ isSignedUpOrDemo = function () {
   return false;
 }
 
+var calculateReferralBonus = function(accountId, plan) {
+  // This function returns an object of the form:
+  //
+  // - {grains: 0, storage: 0}
+  //
+  // which are extra resources this account gets as part of participating in the referral
+  // program. (Storage is measured in bytes, as usual for plans.)
+
+
+  // Authorization note: Only call this if accountId is the current user!
+  var isPaid = (plan && plan !== "free");
+
+  successfulReferralsCount = countReferrals(accountId);
+  if (isPaid) {
+    var maxPaidStorageBonus = 30 * 1e9;
+    return {grains: 0,
+            storage: Math.min(
+              successfulReferralsCount * 2 * 1e9,
+              maxPaidStorageBonus)};
+  } else {
+    var maxFreeStorageBonus = 2 * 1e9;
+    return {grains: successfulReferralsCount * Infinity,
```

Comment:
Zero times Infinity is NaN. I'm guessing you actually wanted `grains` to come out zero when there are no successful referrals. You'll need to use a ternary operator for that.